daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/NullInfosetInputter.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.infoset
+
+import java.io.InputStream
+
+import scala.collection.mutable.ArrayBuffer
+
+import scala.xml.Elem
+import scala.xml.Node
+import scala.xml.SAXParser
+import scala.xml.Text
+import scala.xml.XML
+
+import org.apache.daffodil.dpath.NodeInfo
+import org.apache.daffodil.infoset.InfosetInputterEventType._
+import org.apache.daffodil.util.MaybeBoolean
+import org.apache.daffodil.xml.DaffodilSAXParserFactory
+import org.apache.daffodil.xml.XMLUtils
+
+object NullInfosetInputter {
+
+  case class Event(
+    eventType: InfosetInputterEventType,
+    localName: String = null,
```

Comment:
Can we exclude the overhead of nextElementErd() by timing it separately allowing us to include it, or exclude it from what we're counting? 

The fact that the serialized XML or JSON or whatever doesn't exactly identify the ERD for us is overhead related to the serialized form. (If we invented our own serialized form we would put an unambiguous ERD identifier in the serialized form, so that there would be no nextElementErd() overhead. )

Reply:
I've looked at this more to see if it's even possible, and I'm not sure it's possible without pretty big changes. An `InfosetInputter` implementation doesn't currently know anything about ERDs. And the `InfosetInputter` base class only learns about the root ERD when `initialize()` is called, which happens at the beginning of unparse. So there is no way for an InfosetInputter to know about any ERDs prior to unparsing, and so there's no way to figure out all the information needed for `nextElementERD`.

If we really wanted a super minimal `InfosetInputter`, we would have to do as you suggest and create a custom `InfosetOutputter` that serializes the infoset along with all ERD information, and then a custom InfosetInputter could use that to recreate events and avoid `nextElementErd` overhead. But the feels like quite a bit of work to just exclude the overhead of `nextElementErd()`. And since `nextElementErd()` isn't specific to any `InfosetInputter` implementation, I'd rather we just add instrumentation or profile it separately and find ways to optimize it, rather than creating a special `InfosetInputter`/`Outputter` that avoids it entirely.